core/src/components/skeleton-text/skeleton-text.tsx

```diff
@@ -18,6 +19,24 @@ export class SkeletonText implements ComponentInterface {
    */
   @Prop() animated = false;
 
+  /**
+   * Emitted when the styles change.
+   * @internal
+   */
```

Comment:
Why is this needed now?

Reply:
An `ion-label` containing `ion-skeleton-text` needs to have `overflow: hidden`, otherwise it collapses the margins between the elements:

| `overflow: visible` | `overflow: hidden` |
| ---| ---|
| ![overflow: visible](https://github.com/ionic-team/ionic-framework/assets/6577830/9492ccf7-7a30-4d83-9d80-47568c00703d) | ![overflow: hidden](https://github.com/ionic-team/ionic-framework/assets/6577830/7bce191b-e566-440e-bf30-cc178ae9dfe7) |

When I added the `overflow: hidden` to all labels in an item, it resulted in this weird wrapping behavior:

| | |
| ---| ---|
| ![item overflow hidden](https://github.com/ionic-team/ionic-framework/assets/6577830/e7fdd12d-2df9-4411-a8e6-7c3e5948cb38) | ![checkbox overflow hidden](https://github.com/ionic-team/ionic-framework/assets/6577830/6ec977ce-477b-4878-a02d-5f1ea927cadb) |

I attempted to solve this for `skeleton-text` by adding `display: inline-block` when it is inside of an item and label: https://github.com/ionic-team/ionic-framework/pull/28289/commits/6fdccc4cdf1f157907651b7e199935d0a2228d61 https://github.com/ionic-team/ionic-framework/pull/28289/commits/a00f4f5318b1a7c3a1b17baa7568cb05b32f5198

But that resulted in the thumbnail and avatar alignment being off. Ultimately I decided that passing a class up to item to let it know it contained a `skeleton-text` and style the label that way was the best solution. In addition I kept `overflow: hidden` for legacy items because otherwise the text doesn't ellipsis: 

| `overflow: visible` | `overflow: hidden` |
| ---| ---|
| ![overflow: visible](https://github.com/ionic-team/ionic-framework/assets/6577830/6d013e9a-879a-4834-a25a-ae794e96f205) | ![overflow: hidden](https://github.com/ionic-team/ionic-framework/assets/6577830/7722ba98-6965-4286-8804-0f6b30dbbc38) |

I tried to add comments explaining why specific CSS was added, but lmk if you think there are places I can further clarify!

Reply:
Thanks! Can we add a comment in the JSX file too?

Reply:
Thanks! Can we add a comment in the JSX file too?